### PR TITLE
Minstacks Version 2

### PR DESCRIPTION
**Pull request by AndyTWF**
- Adds websocket functionality to allow the API to push data out to plugins
- Adds websocket event handling for minimum stack level updates
- Adds configurable minimum stack levels so users can choose what they want to see
- Configurable minimum stack levels now includes on a per-airfield basis
- Adds msl-based holding restrictions to hold manager
- Adds active runway tracking

### CODE REVIEW

File: src/sectorfile/RunwayCollection.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                ) const;
+
+                // Inherited via AsrEventHandlerInterface
+                void AsrLoadedEvent(UKControllerPlugin::Euroscope::UserSetting & userSetting) override;
+                void AsrClosingEvent(UKControllerPlugin::Euroscope::UserSetting & userSetting) override;
+
+                // Inherited via RunwayDialogAwareInterface
+                void RunwayDialogSaved(void) override;
+
+                const UKControllerPlugin::SectorFile::Runway invalidRunway = {"", "", 371, false, false};
```
Comment (AxonC) [inline src/sectorfile/RunwayCollection.h]: What does the 371 represent?

File: src/sectorfile/RunwayCollection.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+using UKControllerPlugin::Euroscope::UserSetting;
+using UKControllerPlugin::Euroscope::EuroscopeSectorFileElementInterface;
+using UKControllerPlugin::SectorFile::SectorFileProviderInterface;
+
+namespace UKControllerPlugin {
+    namespace SectorFile {
+
+
+
+        RunwayCollection::RunwayCollection(SectorFileProviderInterface & sectorFile)
```
Comment (AxonC) [inline src/sectorfile/RunwayCollection.cpp]: Could this just be a default constructor `=default` in the header and calling the parent ? Or probably a `= {}`?

File: src/websocket/WebsocketSubscription.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        const std::string WebsocketSubscription::SUB_TYPE_EVENT = "event";
+
+        bool WebsocketSubscription::operator==(const WebsocketSubscription & compare) const
+        {
+            return this->subType == compare.subType && this->subTarget == compare.subTarget;
+        }
+
+        bool WebsocketSubscription::operator<(const WebsocketSubscription & compare) const
+        {
+
```
Comment (AxonC) [inline src/websocket/WebsocketSubscription.cpp]: Whitespace ;)

File: src/sectorfile/RunwayCollection.h (hunk 1 above)
Comment (AndyTWF) [reply src/sectorfile/RunwayCollection.h]: An impossible number of degrees :D

File: src/websocket/WebsocketSubscription.cpp (hunk 3 above)
Comment (AndyTWF) [reply src/websocket/WebsocketSubscription.cpp]: Fixed!

File: src/sectorfile/RunwayCollection.cpp (hunk 2 above)
Comment (AndyTWF) [reply src/sectorfile/RunwayCollection.cpp]: I dont think so?

Comment (AndyTWF) [reply src/sectorfile/RunwayCollection.cpp]: I dont think so?

**Pull request merged**